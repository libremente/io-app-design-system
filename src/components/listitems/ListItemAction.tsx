import React, { useCallback } from "react";
import {
  GestureResponderEvent,
  Pressable,
  StyleSheet,
  Text,
  View
} from "react-native";
import Animated, {
  Extrapolate,
  interpolate,
  interpolateColor,
  useAnimatedStyle,
  useDerivedValue,
  useSharedValue,
  withSpring
} from "react-native-reanimated";
import {
  IOColors,
  IOListItemStyles,
  IOListItemVisualParams,
  IOScaleValues,
  IOSpringValues,
  IOStyles,
  hexToRgba,
  useIOExperimentalDesign,
  useIOTheme
} from "../../core";
import { makeFontStyleObject } from "../../utils/fonts";
import { WithTestID } from "../../utils/types";
import { AnimatedIcon, IOIcons, Icon } from "../icons";
import { buttonTextFontSize } from "../typography/ButtonText";

export type ListItemAction = WithTestID<{
  label: string;
  variant: "primary" | "danger";
  onPress: (event: GestureResponderEvent) => void;
  icon?: IOIcons;
  // Accessibility
  accessibilityLabel: string;
}>;

const styles = StyleSheet.create({
  label: {
    fontSize: buttonTextFontSize,
    lineHeight: 20,
    ...makeFontStyleObject("Regular", false, "ReadexPro")
  }
});

// TODO: Remove this when legacy look is deprecated https://pagopa.atlassian.net/browse/IOPLT-153
const legacyStyles = StyleSheet.create({
  labelLegacy: {
    fontSize: buttonTextFontSize,
    lineHeight: 20,
    ...makeFontStyleObject("SemiBold", false, "TitilliumWeb")
  }
});

export const ListItemAction = ({
  variant,
  label,
  onPress,
  icon,
  accessibilityLabel,
  testID
}: ListItemAction) => {
  const isPressed = useSharedValue(0);

  const { isExperimental } = useIOExperimentalDesign();
  const theme = useIOTheme();

  const mapBackgroundStates: Record<string, string> = {
    default: hexToRgba(IOColors[theme["listItem-pressed"]], 0),
    pressed: IOColors[theme["listItem-pressed"]]
  };

  const mapLegacyForegroundColor: Record<
    NonNullable<ListItemAction["variant"]>,
    IOColors
  > = {
    primary: "blue",
    danger: "error-850"
  };

  const mapForegroundColor: Record<
    NonNullable<ListItemAction["variant"]>,
    string
  > = {
    primary: IOColors[theme["interactiveElem-default"]],
    danger: IOColors[theme.errorText]
  };

  // TODO: Remove this when legacy look is deprecated https://pagopa.atlassian.net/browse/IOPLT-153
  const legacyItemActionIcon = (icon: IOIcons) => (
    <Icon
      name={icon}
      color={mapLegacyForegroundColor[variant]}
      size={IOListItemVisualParams.iconSize}
    />
  );

  const itemActionIcon = (icon: IOIcons) => (
    <>
      <AnimatedIcon
        name={icon}
        color={mapForegroundColor[variant] as IOColors}
        size={IOListItemVisualParams.iconSize}
      />
    </>
  );

  const itemActionIconComponent = (icon: IOIcons) =>
    isExperimental ? itemActionIcon(icon) : legacyItemActionIcon(icon);

  const textStyle = [styles.label, { color: mapForegroundColor[variant] }];

  const legacyTextStyle = [
    legacyStyles.labelLegacy,
    { color: IOColors[mapLegacyForegroundColor[variant]] }
  ];

  const textStyleComponent = isExperimental ? textStyle : legacyTextStyle;

  // Scaling transformation applied when the button is pressed
  const animationScaleValue = IOScaleValues?.basicButton?.pressedState;

  const progressPressed = useDerivedValue(() =>
    withSpring(isPressed.value, IOSpringValues.button)
  );

  // Interpolate animation values from `isPressed` values
  const animatedScaleStyle = useAnimatedStyle(() => {
    const scale = interpolate(
      progressPressed.value,
      [0, 1],
      [1, animationScaleValue],
      Extrapolate.CLAMP
    );

    return {
      transform: [{ scale }]
    };
  });

  const animatedBackgroundStyle = useAnimatedStyle(() => {
    const backgroundColor = interpolateColor(
      progressPressed.value,
      [0, 1],
      [mapBackgroundStates.default, mapBackgroundStates.pressed]
    );

    return {
      backgroundColor
    };
  });

  const handlePressIn = useCallback(() => {
    // eslint-disable-next-line functional/immutable-data
    isPressed.value = 1;
  }, [isPressed]);
  const handlePressOut = useCallback(() => {
    // eslint-disable-next-line functional/immutable-data
    isPressed.value = 0;
  }, [isPressed]);

  return (
    <Pressable
      onPress={onPress}
      onPressIn={handlePressIn}
      onPressOut={handlePressOut}
      onTouchEnd={handlePressOut}
      accessible={true}
      accessibilityLabel={accessibilityLabel}
      accessibilityRole="button"
      testID={testID}
    >
      <Animated.View
        style={[IOListItemStyles.listItem, animatedBackgroundStyle]}
      >
        <Animated.View
          style={[IOListItemStyles.listItemInner, animatedScaleStyle]}
        >
          {icon && (
            <View style={{ marginRight: IOListItemVisualParams.iconMargin }}>
              {itemActionIconComponent(icon)}
            </View>
          )}
          <View style={IOStyles.flex}>
            <Text style={textStyleComponent}>{label}</Text>
          </View>
        </Animated.View>
      </Animated.View>
    </Pressable>
  );
};

export default ListItemAction;
