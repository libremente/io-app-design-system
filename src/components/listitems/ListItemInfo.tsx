import React from "react";
import { View } from "react-native";
import {
  IOListItemStyles,
  IOListItemVisualParams,
  IOStyles,
  useIOTheme
} from "../../core";
import { WithTestID } from "../../utils/types";
import { Icon, IOIcons } from "../icons";
import { H6, LabelSmall } from "../typography";

export type ListItemInfo = WithTestID<{
  label: string;
  value: string | React.ReactNode;
  numberOfLines?: number;
  icon?: IOIcons;
  // Accepted components: ButtonLink, IconButton
  // Don't use any components other than these
  action?: React.ReactNode;
  // Accessibility
  accessibilityLabel: string;
}>;

export const ListItemInfo = ({
  label,
  value,
  numberOfLines = 2,
  icon,
  action,
  accessibilityLabel,
  testID
}: ListItemInfo) => {
  const theme = useIOTheme();
  return (
    <View
      style={IOListItemStyles.listItem}
      testID={testID}
      accessible={true}
      accessibilityLabel={accessibilityLabel}
    >
      <View style={IOListItemStyles.listItemInner}>
        {icon && (
          <View style={{ marginRight: IOListItemVisualParams.actionMargin }}>
            <Icon
              name={icon}
              color="grey-450"
              size={IOListItemVisualParams.iconSize}
            />
          </View>
        )}
        <View style={IOStyles.flex}>
          <LabelSmall weight="Regular" color={theme["textBody-tertiary"]}>
            {label}
          </LabelSmall>
          <H6 color={theme["textBody-default"]} numberOfLines={numberOfLines}>
            {value}
          </H6>
        </View>
        {action && (
          <View style={{ marginLeft: IOListItemVisualParams.actionMargin }}>
            {action}
          </View>
        )}
      </View>
    </View>
  );
};

export default ListItemInfo;
