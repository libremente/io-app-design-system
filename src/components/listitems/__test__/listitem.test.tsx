import React from "react";
import { Alert } from "react-native";
import * as TestRenderer from "react-test-renderer";
import ListItemInfo from "../ListItemInfo";
import ListItemInfoCopy from "../ListItemInfoCopy";
import ListItemNav from "../ListItemNav";
import ListItemNavAlert from "../ListItemNavAlert";
import ListItemAction from "../ListItemAction";
import ListItemIDP from "../ListItemIDP";
import { ListItemTransaction } from "../ListItemTransaction";
import { PressableListItemBase } from "../PressableListItemsBase";

const onButtonPress = () => {
  Alert.alert("Alert", "Action triggered");
};

describe("Test List Item Components", () => {
  it("ListItemInfo Snapshot", () => {
    const listItemInfo = TestRenderer.create(<ListItemInfo label={"label"} value={"testValue"} accessibilityLabel={"accessibilityLabel"}></ListItemInfo>).toJSON();
    expect(listItemInfo).toMatchSnapshot();
  });
  it("ListItemInfoCopy Snapshot", () => {
    const listItemInfoCopy = TestRenderer.create(<ListItemInfoCopy label={"label"} value={"testValue"} onPress={onButtonPress} accessibilityLabel={"accessibilityLabel"}></ListItemInfoCopy>).toJSON();
    expect(listItemInfoCopy).toMatchSnapshot();
  });
  it("ListItemNav Snapshot", () => {
    const listItemNav = TestRenderer.create(<ListItemNav value={"testValue"} onPress={onButtonPress} accessibilityLabel={"accessibilityLabel"} ></ListItemNav>).toJSON();
    expect(listItemNav).toMatchSnapshot();
  });
  it("ListItemNavAlert Snapshot", () => {
    const listItemNavAlert = TestRenderer.create(<ListItemNavAlert value={"testValue"} onPress={onButtonPress} accessibilityLabel={"accessibilityLabel"} ></ListItemNavAlert>).toJSON();
    expect(listItemNavAlert).toMatchSnapshot();
  });
  it("ListItemAction Snapshot", () => {
    const listItemAction = TestRenderer.create(<ListItemAction label={"label"} variant={"primary"} onPress={onButtonPress} accessibilityLabel={"accessibilityLabel"}></ListItemAction>).toJSON();
    expect(listItemAction).toMatchSnapshot();
  });
  it("ListItemIDP Snapshot", () => {
    const listItemIDP = TestRenderer.create(<ListItemIDP></ListItemIDP>).toJSON();
    expect(listItemIDP).toMatchSnapshot();
  });
  it("ListItemTransaction Snapshot", () => {
    const listItemIDP = TestRenderer.create(<ListItemTransaction
      title="TITLE"
      subtitle="subtitle"
      transactionStatus="success"
      transactionAmount="â‚¬ 1.000,00"
      isLoading={true}
      onPress={onButtonPress}
    />).toJSON();
    expect(listItemIDP).toMatchSnapshot();
  });
  it("PressableListItemsBase Snapshot", () => {
    const listItemIDP = TestRenderer.create(<PressableListItemBase></PressableListItemBase>).toJSON();
    expect(listItemIDP).toMatchSnapshot();
  });
});
