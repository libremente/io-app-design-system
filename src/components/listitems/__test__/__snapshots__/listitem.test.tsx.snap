// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test List Item Components ListItemAction Snapshot 1`] = `
<View
  accessibilityLabel="accessibilityLabel"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTouchEnd={[Function]}
>
  <View
    style={
      [
        {
          "marginHorizontal": -24,
          "paddingHorizontal": 24,
          "paddingVertical": 12,
        },
        {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          {
            "transform": [
              {
                "scale": undefined,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Titillium Web",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "600",
                "lineHeight": 20,
              },
              {
                "color": "#0073E6",
              },
            ]
          }
        >
          label
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test List Item Components ListItemIDP Snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTouchEnd={[Function]}
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E8EBF1",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        },
        false,
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "alignSelf": "center",
            "color": "#555C70",
            "flexShrink": 1,
            "fontSize": 12,
            "lineHeight": 16,
            "textTransform": "uppercase",
          },
          {
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
    />
    <Image
      source={
        {
          "uri": "undefined",
        }
      }
      style={
        {
          "height": 30,
          "marginStart": 8,
          "resizeMode": "contain",
          "width": 120,
        }
      }
    />
  </View>
</View>
`;

exports[`Test List Item Components ListItemInfo Snapshot 1`] = `
<View
  accessibilityLabel="accessibilityLabel"
  accessible={true}
  style={
    {
      "marginHorizontal": -24,
      "paddingHorizontal": 24,
      "paddingVertical": 12,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        accessible={true}
        importantForAccessibility="yes"
      >
        <Text
          allowFontScaling={false}
          color="grey-700"
          defaultColor="blue"
          defaultWeight="Bold"
          font="TitilliumWeb"
          fontStyle={
            {
              "fontSize": 14,
              "lineHeight": 21,
            }
          }
          style={
            [
              {
                "fontSize": 14,
                "lineHeight": 21,
              },
              {
                "color": "#555C70",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "400",
              },
            ]
          }
          weight="Regular"
        >
          label
        </Text>
        <Text
          allowFontScaling={false}
          color="black"
          defaultColor="black"
          defaultWeight="SemiBold"
          font="TitilliumWeb"
          fontStyle={
            {
              "fontSize": 18,
              "lineHeight": 25,
            }
          }
          numberOfLines={2}
          style={
            [
              {
                "fontSize": 18,
                "lineHeight": 25,
              },
              {
                "color": "#0E0F13",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
        >
          testValue
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test List Item Components ListItemInfoCopy Snapshot 1`] = `
<View
  accessibilityLabel="accessibilityLabel"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      [
        {
          "marginHorizontal": -24,
          "paddingHorizontal": 24,
          "paddingVertical": 12,
        },
        {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          {
            "transform": [
              {
                "scale": undefined,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="grey-700"
          defaultColor="blue"
          defaultWeight="Bold"
          font="TitilliumWeb"
          fontStyle={
            {
              "fontSize": 14,
              "lineHeight": 21,
            }
          }
          style={
            [
              {
                "fontSize": 14,
                "lineHeight": 21,
              },
              {
                "color": "#555C70",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "400",
              },
            ]
          }
          weight="Regular"
        >
          label
        </Text>
        <Text
          allowFontScaling={false}
          color="blue"
          defaultColor="black"
          defaultWeight="SemiBold"
          font="TitilliumWeb"
          fontStyle={
            {
              "fontSize": 18,
              "lineHeight": 25,
            }
          }
          numberOfLines={2}
          style={
            [
              {
                "fontSize": 18,
                "lineHeight": 25,
              },
              {
                "color": "#0073E6",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
        >
          testValue
        </Text>
      </View>
      <View
        style={
          {
            "marginLeft": 16,
          }
        }
      >
        <RNSVGSvgView
          accessibilityElementsHidden={true}
          accessibilityLabel=""
          accessible={false}
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          color={4278219750}
          focusable={false}
          height={24}
          importantForAccessibility="no-hide-descendants"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "#0073E6",
              },
              {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          tintColor={4278219750}
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup>
            <RNSVGPath
              clipRule={0}
              d="M12 18h8a2 2 0 0 0 2-2V9h-4a3 3 0 0 1-3-3V2h-3a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2Zm9.332-11.987a2 2 0 0 1 .603.987H18a1 1 0 0 1-1-1V2.227a2 2 0 0 1 .406.281l3.926 3.505ZM14 20h-2a4 4 0 0 1-4-4V6H4a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h8a2 2 0 0 0 2-2Zm2 0h4a4 4 0 0 0 4-4V7.505a4 4 0 0 0-1.336-2.984l-3.926-3.505A4 4 0 0 0 16.074 0H12a4 4 0 0 0-4 4H4a4 4 0 0 0-4 4v12a4 4 0 0 0 4 4h8a4 4 0 0 0 4-4Z"
              fill={
                [
                  2,
                ]
              }
              fillRule={0}
              propList={
                [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test List Item Components ListItemNav Snapshot 1`] = `
<View
  accessibilityLabel="accessibilityLabel"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      [
        {
          "marginHorizontal": -24,
          "paddingHorizontal": 24,
          "paddingVertical": 12,
        },
        {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          {
            "transform": [
              {
                "scale": undefined,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="black"
          defaultColor="black"
          defaultWeight="SemiBold"
          font="TitilliumWeb"
          fontStyle={
            {
              "fontSize": 18,
              "lineHeight": 25,
            }
          }
          style={
            [
              {
                "fontSize": 18,
                "lineHeight": 25,
              },
              {
                "color": "#0E0F13",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
        >
          testValue
        </Text>
      </View>
      <View
        style={
          {
            "marginLeft": 16,
          }
        }
      >
        <RNSVGSvgView
          accessibilityElementsHidden={true}
          accessibilityLabel=""
          accessible={false}
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          color={4278219750}
          focusable={false}
          height={24}
          importantForAccessibility="no-hide-descendants"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "#0073E6",
              },
              {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          tintColor={4278219750}
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup>
            <RNSVGPath
              clipRule={0}
              d="M16.2929 5.29289c.3905-.39052 1.0237-.39052 1.4142 0l6 6.00001c.3905.3905.3905 1.0237 0 1.4142l-6 6c-.3905.3905-1.0237.3905-1.4142 0-.3905-.3905-.3905-1.0237 0-1.4142L21.5858 12l-5.2929-5.29289c-.3905-.39053-.3905-1.02369 0-1.41422Z"
              fill={
                [
                  2,
                ]
              }
              fillRule={0}
              propList={
                [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test List Item Components ListItemNavAlert Snapshot 1`] = `
<View
  accessibilityLabel="accessibilityLabel"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      [
        {
          "marginHorizontal": -24,
          "paddingHorizontal": 24,
          "paddingVertical": 12,
        },
        {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          {
            "transform": [
              {
                "scale": undefined,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          {
            "marginRight": 16,
          }
        }
      >
        <RNSVGSvgView
          accessibilityElementsHidden={true}
          accessibilityLabel=""
          accessible={false}
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          color={4294862438}
          focusable={false}
          height={24}
          importantForAccessibility="no-hide-descendants"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "#FE6666",
              },
              {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          tintColor={4294862438}
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup>
            <RNSVGPath
              clipRule={0}
              d="M7.00702 1c-.73494 0-1.41405.39209-1.78152 1.02856L.2756 11.0274c-.36747.6365-.36747 1.4206 0 2.0571l4.9499 8.8783c.36747.6365 1.04658 1.0286 1.78152 1.0286h9.94288c.7349 0 1.414-.3921 1.7815-1.0286l4.993-8.8783c.3675-.6365.3675-1.4206 0-2.0571l-4.993-8.99884C18.3639 1.39209 17.6848 1 16.9499 1H7.00702Zm5.02608 13.5549c.5676 0 1.0283-.4601 1.0283-1.0277V5.52583c0-.56758-.4601-1.02769-1.0277-1.02769-.5676 0-1.0282.46011-1.0282 1.02769v8.00137c0 .5676.4601 1.0277 1.0276 1.0277Zm0 1.9431c.8436 0 1.5275.6839 1.5275 1.5275 0 .8436-.6839 1.5275-1.5275 1.5275-.8436 0-1.5274-.6839-1.5274-1.5275 0-.8436.6838-1.5275 1.5274-1.5275Z"
              fill={
                [
                  2,
                ]
              }
              fillRule={0}
              propList={
                [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="black"
          defaultColor="black"
          defaultWeight="SemiBold"
          font="TitilliumWeb"
          fontStyle={
            {
              "fontSize": 18,
              "lineHeight": 25,
            }
          }
          style={
            [
              {
                "fontSize": 18,
                "lineHeight": 25,
              },
              {
                "color": "#0E0F13",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
        >
          testValue
        </Text>
      </View>
      <View
        style={
          {
            "marginLeft": 16,
          }
        }
      >
        <RNSVGSvgView
          accessibilityElementsHidden={true}
          accessibilityLabel=""
          accessible={false}
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          color={4278219750}
          focusable={false}
          height={24}
          importantForAccessibility="no-hide-descendants"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "#0073E6",
              },
              {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          tintColor={4278219750}
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup>
            <RNSVGPath
              clipRule={0}
              d="M16.2929 5.29289c.3905-.39052 1.0237-.39052 1.4142 0l6 6.00001c.3905.3905.3905 1.0237 0 1.4142l-6 6c-.3905.3905-1.0237.3905-1.4142 0-.3905-.3905-.3905-1.0237 0-1.4142L21.5858 12l-5.2929-5.29289c-.3905-.39053-.3905-1.02369 0-1.41422Z"
              fill={
                [
                  2,
                ]
              }
              fillRule={0}
              propList={
                [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test List Item Components ListItemTransaction Snapshot 1`] = `
<View
  accessible={false}
  style={
    {
      "marginHorizontal": -24,
      "paddingHorizontal": 24,
      "paddingVertical": 12,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "marginRight": 16,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "opacity": 0.5,
          }
        }
      >
        <View
          height={44}
          radius={100}
          style={
            {
              "backgroundColor": "#efefef",
              "borderRadius": 100,
              "height": 44,
              "width": 44,
            }
          }
          width={44}
        />
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "opacity": 0.5,
          }
        }
      >
        <View
          height={16}
          radius={8}
          style={
            {
              "backgroundColor": "#efefef",
              "borderRadius": 8,
              "height": 16,
              "width": 62,
            }
          }
          width={62}
        />
      </View>
      <View
        style={
          {
            "height": 4,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "opacity": 0.5,
          }
        }
      >
        <View
          height={16}
          radius={8}
          style={
            {
              "backgroundColor": "#efefef",
              "borderRadius": 8,
              "height": 16,
              "width": 107,
            }
          }
          width={107}
        />
      </View>
    </View>
    <View
      style={
        {
          "marginLeft": 16,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "opacity": 0.5,
          }
        }
      >
        <View
          height={24}
          radius={8}
          style={
            {
              "backgroundColor": "#efefef",
              "borderRadius": 8,
              "height": 24,
              "width": 70,
            }
          }
          width={70}
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Test List Item Components PressableListItemsBase Snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTouchEnd={[Function]}
>
  <View
    style={
      [
        {
          "marginHorizontal": -24,
          "paddingHorizontal": 24,
          "paddingVertical": 12,
        },
        {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          {
            "transform": [
              {
                "scale": undefined,
              },
            ],
          },
        ]
      }
    />
  </View>
</View>
`;
